src/components/OntologyBrowser/index.tsx

```diff
@@ -123,16 +165,22 @@ class OntologyModal extends React.Component<ModalProps, ModalState> {
     });
   };
 
-  updateData = () => {
-    this.ontologyStore.fetch(this.props.initialSqon).then(() => {
+  updateData = async () => {
+    this.setState({ isLoading: true });
+    const { initialSqon } = this.props;
+    try {
+      await this.ontologyStore.fetch(initialSqon);
```

Comment:
same thing here, if the props is only use one time on a short line, maybe desconstructing it add no or little value. What do you think?

Reply:
What you say makes sense but I kept the same 'style' (deconstructing) everywhere in the file. But I'll do what you suggest.